src/visitors.rs

```diff
@@ -1098,12 +1093,54 @@ impl<'a, 'b: 'a, 'tcx: 'b> MirVisitor<'a, 'b, 'tcx> {
         right_operand: &mir::Operand,
     ) {
         debug!("default visit_checked_binary_op(path: {:?}, bin_op: {:?}, left_operand: {:?}, right_operand: {:?})", path, bin_op, left_operand, right_operand);
-        let _left = self.visit_operand(left_operand);
-        let _right = self.visit_operand(right_operand);
-        //todo: get a value that is the checked bin_op of _left and _right.
-        //todo: what should happen if the operation overflows?
+        // We assume that path is a temporary used to track the operation result and its overflow status.
+        let target_type = match path {
+            Path::LocalVariable { ordinal } => {
+                let loc = &self.mir.local_decls[mir::Local::from(ordinal)];
+                match loc.ty.sty {
+                    TyKind::Tuple(types) => (&types[0].sty).into(),
+                    _ => unreachable!(),
```

Comment:
if let TyKind::Tuple(types) = loc.ty.sty {
   ...
} else {
  unreachable!()
}

is arguably more readable. It's a toss up, do what makes you happy.

Reply:
I did think about using if let, but match seemed more readable to me at the time.